### ISSUE TITLE
Incorporate uncertainty on non Bayesian forecasting

### ISSUE THREAD

**Issue by jarroyoe** [Jack-H-Buckner/UniversalDiffEq.jl#40]
I went deep through the rabbit hole of uncertainty quantification in forecasting of neural networks. A faster method of quantifying uncertainty than what was proposed by [Lapeyrolerie et al, 2022](https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.13954) of fitting several neural networks and doing an ensemble forecast is to do a [Laplace approximation](https://arxiv.org/abs/2106.14806) of the neural network. This has been [tested for NODEs](https://arxiv.org/abs/2305.13290) and implemented in Julia as [LaplaceRedux.jl](https://github.com/JuliaTrustworthyAI/LaplaceRedux.jl).

The idea of Laplace approximation is to linearize the neural network from  the Maximum a Posteriori estimation of the parameters. Ott et al proposes this linearization to occur using the final iteration of SGD as the MAP. For BayesianUDEs (see https://github.com/Jack-H-Buckner/UniversalDiffEq.jl/issues/38), we can choose the mean of each parameter as the MAP, but that would assume independence on the posterior distribution.

Thoughts?

**Comment by jarroyoe** [comment]
[This paper](https://arxiv.org/abs/2406.08853) was recently submitted to arxiv, might be worth considering.

**Comment by gustavdelius** [comment]
It may be worth adding that the Laplace approximation would be useful not only in order to get uncertainty estimates but also in order to avoid bias in your parameter estimation. The correct (unbiased) way to estimate the model parameters in a state space model is to use the marginal loss function obtained by integrating out the state variables. The Laplace approximation is a good way for doing that.
